ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -208,6 +224,17 @@ public void handlePost(RestRequest restRequest, RestResponseChannel restResponse
       logger.trace("Handling POST request - {}", restRequest.getUri());
       checkAvailable();
       long propsBuildStartTime = System.currentTimeMillis();
+      accountAndContainerSanityCheck(restRequest);
+      // check if the request is old (with no accountName nor containerName) or new (otherwise)
```

Comment:
will discuss offline with you if this can be refactored. Prefer to have minimal functionality in this part of the function to make it easier to read.